### DISCUSSION TITLE
How to combine multiple custom hooks into one?

### DISCUSSION THREAD

**Discussion by BiosSun** [TanStack/query#2492] [Q&A]
for example:

```typescript
function useCurrentUser(): UseQueryResult<User, Error> {
    const userIdQuery = useCurrentUserId();
    const userQuery = useUser(userIdQuery.data);

    // In here, I how to return a combined query result?
}
```

**Comment by vago** [comment]
Wouldn't returning a new object composed of both query's work? Something like this.

````
return {userIdQuery, userQuery}
````

**Comment by TkDodo** [comment]
It depends on your data - you can create a new object and merge those two; or if it's an array you can create a new one an concat them, or merge them... It's not really react-query specific, just javascript. Both queries will return `.data`, which can be potentially undefined or contain your server data, so you can merge them depending on your use case. There is no generic answer to that.

**Reply by BiosSun** [reply]
Oh, Thank you, I get it.
